Use variables for PCoIP Agent download location

Edwin-Pau opened the pull request:
As part of the startup scripts for CentOS machines, the download
locations for PCoIP agent has been changed to use terraform variables
instead for better flexibility and to remain consistent with the
Windows startup scripts. Windows startup scripts were modified
to match the style of the Centos startup scripts using global
variables.

Signed-off-by: Edwin Pau <epau@teradici.com>

Edwin-Pau commented:
In CentOS, it downloads the rpm into /var/cache/yum as the default path before installing, I'm not sure if you wanted this changed instead of just the .repo file.

shermanyin reviewed (changes requested):
Thanks for correcting the logs from these scripts.

Sorry the problem description wasn't very clear.  What I meant was in windows there is the "pcoip_agent_location" variable to indicate which URL to fetch the driver, where as it is hard-coded in CentOS.  So What i was thinking was just makeing "https://downloads.teradici.com/rhel/" a variable that could be changed by terraform user.

Commit:
2/2 GCP: use vars for PCoIP Agent location

As part of the startup scripts for CentOS machines, the download
locations for PCoIP agent has been changed to use terraform variables
for better flexibility and to remain consistent with the Windows
startup scripts. Windows startup scripts were modified to match the
style of the Centos startup scripts using global variables.

Signed-off-by: Edwin Pau <epau@teradici.com>

shermanyin commented on modules/aws/centos-gfx/centos-gfx-startup.sh.tmpl:
```diff
@@ -153,24 +152,24 @@ enable_persistence_mode() {
-        log "--> Get Teradici PCoIP agent repo"
-        wget --retry-connrefused --tries=3 --waitretry=5 -O /etc/yum.repos.d/pcoip.repo https://downloads.teradici.com/rhel/pcoip.repo
+        log "--> Getting Teradici pubkey..."
+        teradici_pubkey=teradici.pub.gpg
```
this variable doesn't seem to accomplish anything since it's not used anywhere else.

shermanyin commented on modules/aws/centos-gfx/vars.tf:
```diff
@@ -100,6 +100,16 @@ variable "admin_ssh_key_name" {
 
+variable "pcoip_agent_location" {
+  description = "URL of Teradici PCoIP Graphics Agent"
+  default     = "https://downloads.teradici.com/rhel/"
```
Given how the agent is installed by downloading a repo and its public key, i think for CentOS it make sense to have a pcoip_agent_repo_public_key and default to https://downloads.teradici.com/rhel/teradici.pub.gpg and a pcoip_agent_repo_url and default to https://downloads.teradici.com/rhel/pcoip.repo

shermanyin commented on modules/aws/centos-gfx/centos-gfx-startup.sh.tmpl:
```diff
@@ -153,24 +152,24 @@ enable_persistence_mode() {
 
-        log "--> Install PCoIP graphics agent ..."
+        log "--> Getting Teradici PCoIP agent repo..."
+        wget --retry-connrefused --tries=3 --waitretry=5 -O /etc/yum.repos.d/pcoip.repo $PCOIP_AGENT_LOCATION$PCOIP_AGENT_REPO
```
can we just specify the output directory to be in /etc/yum.repos.d/ and then the file should be unchanged from where it is downloaded from? that way we don't have to hard code "pcoip.repo".

shermanyin commented on modules/aws/win-gfx/win-gfx-startup.ps1.tmpl:
```diff
@@ -73,18 +75,20 @@ function PCoIP-Agent-Install {
-    if ("${pcoip_agent_filename}") {
-        $agent_filename = "${pcoip_agent_filename}"
+    if (![string]::IsNullOrEmpty($PCOIP_AGENT_FILENAME)) {
+        "Using custom PCoIP Agent filename..."
```
custom sounds funny. maybe user-specified?

shermanyin reviewed (changes requested):
Step in the right direction, however I would like to still see some changes.

And since you will be creating a new patchset anyway, could you move all the comments / logs / typo changes to one separate commit (no functional difference), and the parameterization of the repo/key a separate commit?

Commit:
1/2 AWS: use vars for PCoIP Agent location

As part of the startup scripts for CentOS machines, the download
locations for PCoIP agent has been changed to use terraform variables
for better flexibility and to remain consistent with the Windows
startup scripts. Windows startup scripts were modified to match the
style of the Centos startup scripts using global variables.

Signed-off-by: Edwin Pau <epau@teradici.com>

Edwin-Pau commented:
- Resolved all the comments above
- Removed all logging style, grammar, spelling changes
- Tested all 8 workstation types with the client (GCP + AWS)
